File: codegen/results/v001/qwen3.5_27b_think/20260408_134639/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_tests():
    # Test cases defined as tuples: (price, quantity, expected_result, description)
    # Using price = 10.0 for easy calculation verification
    test_cases = [
        (10.0, 9, 90.0, "Quantity 9 (No discount)"),
        (10.0, 10, 90.0, "Quantity 10 (10% off boundary)"),
        (10.0, 49, 441.0, "Quantity 49 (10% off boundary)"),
        (10.0, 50, 375.0, "Quantity 50 (25% off boundary)"),
    ]

    for price, quantity, expected, description in test_cases:
        result = calculate_discount(price, quantity)
        
        # Use a small epsilon for floating point comparison safety
        if abs(result - expected) < 1e-9:
            print(f"PASS: {description}")
        else:
            # Format floats to 2 decimal places for clean output
            print(f"FAIL: {description} (expected: {expected:.2f}, got: {result:.2f})")


if __name__ == "__main__":
    run_tests()

File: codegen/results/v001/qwen3.5_27b_think/20260408_134639/typescript/005_unit_test_writer/solution.ts
// Original function to test
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Test runner
interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

const testCases: TestCase[] = [
    {
        description: "No discount for quantity 9",
        price: 100,
        quantity: 9,
        expected: 900
    },
    {
        description: "10% discount for quantity 10 (boundary)",
        price: 100,
        quantity: 10,
        expected: 900
    },
    {
        description: "10% discount for quantity 49 (boundary)",
        price: 100,
        quantity: 49,
        expected: 4410
    },
    {
        description: "25% discount for quantity 50 (boundary)",
        price: 100,
        quantity: 50,
        expected: 3750
    },
    {
        description: "Large quantity with 25% discount",
        price: 200,
        quantity: 100,
        expected: 15000
    }
];

// Run tests
let passCount = 0;
let failCount = 0;

for (const test of testCases) {
    const result = calculateDiscount(test.price, test.quantity);
    // Use a small epsilon for floating point comparison
    const epsilon = 0.0001;
    
    if (Math.abs(result - test.expected) < epsilon) {
        console.log(`PASS: ${test.description}`);
        passCount++;
    } else {
        console.log(`FAIL: ${test.description} (expected: ${test.expected}, got: ${result})`);
        failCount++;
    }
}

console.log(`\nResults: ${passCount} passed, ${failCount} failed`);